#全局参数
#------ 高速工况（15~40km/h）------

#PID调参方法
# 比例P↑-----上升时间↓-----超调量↑-----稳态误差  ↓  -----稳定性↓-----安定时间↗
# 积分 I↑-----上升时间↘-----超调量↑-----稳态误差↓↓-----稳定性↓-----安定时间↑
# 微分D↑-----上升时间↘-----超调量↓-----稳态误差   ~   -----稳定性↑-----安定时间↓

HighSpeedParams:
  Lat_P: 2.8 # 2.6
  Lat_I: 0.04 #0.06
  Lat_D: 0.25 #0.12
  Angular_max_abs_for_LS: 2.1 # 低速工况's 最大转向速度的绝对值，必须是正值，且较大。单位：rad/s，低速工况采用该“最大角速度”，中速工况采用衰减的“最大角速度”
  Angular_max_abs_for_HS: 0.3 # 高速工况's 最大转向速度的绝对值，必须是正值，且较小。单位：rad/s，当前0.087=5°/180°*π，高速工况采用该“最大角速度”
  Low_linear_for_HS_w_attenuate: 2.0 #低速工况判据（下发的线速度Linear小于该值认为“低速工况”），较小值，必须为正（本质是绝对值）
  High_linear_for_HS_w_attenuate: 6.0 #高速工况判据（下发的线速度Linear大于该值认为“高速工况”），Linear处于二者之间的，是“中速工况”，较大值，必须为正（本质是绝对值）
  Angular_delta_max: 0.2 #15*M_PI/180

  Lon_P: 0.5 #0.55
  Lon_I: 0.08
  Lon_D: 0.12
  Linear_max: 12.5        # 实际下发到底盘的速度指令の最大值（必须大于DESIRED_LINEAR!!!!!!!!!!）
  Linear_min: 0         #
  Linear_delta_max: 1.0  #2 / 3.6

  DESIRED_LINEAR: 0.888666 #期望的巡航线速度 （必须小于Linear_max!!!!!!!!!!） 单位：m/s #如果不用期望的方法，就用不到。
  DESIRED_MIN_LINEAR: 0.7 #预设的最小的车辆速度（刹车前夕，拐急弯时） 单位：m/s
  d_near: 13.5 #近距工况的距离判断阈值 单位：m #2.0-lowspeed;  11m/s-13.5m  6m/s-8m左右
  d_stop: 1.5 #停车距离  单位：m
  k_ramp: 1 #坡道工况时的期望速度系数
  w_threshold_sharp_turn: 0.9   #急弯w阈值，应该是大值，进行线速度优化需要使用 #1.2
  w_threshold_gentle_turn: 0.6   #缓弯w阈值，应该是小值，进行线速度优化需要使用 #0.6

  Target_L: 0.25 #目标的膨胀尺寸 单位：m 
  Veh_L: 3.2  #车长 单位：m
  PRE_DIS: 3.8 #最小的预瞄距离，根据速度进行向上调节 单位：m
  u_weight: 0.6 #预瞄距离的速度影响系数，pre_dis = PRE_DIS + u_weight*V

